src/main/java/org/embulk/base/restclient/record/JacksonServiceRecord.java

```diff
@@ -3,17 +3,21 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 public class JacksonServiceRecord
-        extends ServiceRecord<JacksonValueLocator>
+        extends ServiceRecord
 {
     public JacksonServiceRecord(ObjectNode record)
     {
         this.record = record;
     }
 
     @Override
-    public JacksonServiceValue getValue(JacksonValueLocator locator)
+    public JacksonServiceValue getValue(ValueLocator locator)
     {
-        return new JacksonServiceValue(locator.locateValue(record));
+        if (locator instanceof JacksonValueLocator) {
+            JacksonValueLocator jacksonLocator = (JacksonValueLocator) locator;
```

Comment:
I concern:
a. The cost of those dynamic type checks. I think that it's really small instead of json value parsing but.
b. Would be much better if the type check is executed at compile-time or before Embulk transaction starts.